src/ci/shared.sh

```diff
@@ -15,6 +15,12 @@
 
 # See http://unix.stackexchange.com/questions/82598
 function retry {
+  # Only do retries on CI, locally they shouldn't really be necessary
+  if [ -z "$TRAVIS$APPVEYOR" ]; then
```

Comment:
Travis complains about AppVeyor here 😄 

```
[00:00:00] /home/travis/build/rust-lang/rust/src/ci/shared.sh: line 19: APPVEYOR: unbound variable
```

You should be able to just check `$CI`, this is defined on both [Travis](https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables) and [AppVeyor](https://www.appveyor.com/docs/environment-variables/). This might need to be manually passed to the Docker instances though.